src/shazamkit.cs

```diff
@@ -0,0 +1,268 @@
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SHCatalog
+	{
+		[Export ("minimumQuerySignatureDuration")]
+		double MinimumQuerySignatureDuration { get; }
+
+		[Export ("maximumQuerySignatureDuration")]
+		double MaximumQuerySignatureDuration { get; }
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (SHCatalog))]
+	interface SHCustomCatalog
+	{
+		[Export ("addReferenceSignature:representingMediaItems:error:")]
+		bool Add (SHSignature signature, SHMediaItem[] mediaItems, [NullAllowed] out NSError error);
+
+		[Export ("addCustomCatalogFromURL:error:")]
+		bool Add (NSUrl url, [NullAllowed] out NSError error);
+
+		[Export ("writeToURL:error:")]
+		bool Write (NSUrl url, [NullAllowed] out NSError error);
+
+		[Static]
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0)]
+		[Export ("new")]
+		[return: Release]
+		SHCustomCatalog Create ();
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SHMatch : NSSecureCoding
+	{
+		[Export ("mediaItems", ArgumentSemantic.Strong)]
+		SHMatchedMediaItem[] MediaItems { get; }
+
+		[Export ("querySignature", ArgumentSemantic.Strong)]
+		SHSignature QuerySignature { get; }
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (SHMediaItem))]
+	[DisableDefaultCtor]
+	interface SHMatchedMediaItem : NSSecureCoding
+	{
+		[Export ("frequencySkew")]
+		float FrequencySkew { get; }
+
+		[Export ("matchOffset")]
+		double MatchOffset { get; }
+
+		[Export ("predictedCurrentMatchOffset")]
+		double PredictedCurrentMatchOffset { get; }
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SHMediaItem : NSSecureCoding, NSCopying
+	{
+		[NullAllowed]
+		[Export ("shazamID")]
+		string ShazamId { get; }
+
+		[NullAllowed]
+		[Export ("title")]
+		string Title { get; }
+
+		[NullAllowed]
+		[Export ("subtitle")]
+		string Subtitle { get; }
+
+		[NullAllowed]
+		[Export ("artist")]
+		string Artist { get; }
+
+		[Export ("genres", ArgumentSemantic.Strong)]
+		string[] Genres { get; }
+
+		[NullAllowed]
+		[Export ("appleMusicID")]
+		string AppleMusicId { get; }
+
+		[NullAllowed]
+		[Export ("appleMusicURL", ArgumentSemantic.Strong)]
+		NSUrl AppleMusicUrl { get; }
+
+		[NullAllowed]
+		[Export ("webURL", ArgumentSemantic.Strong)]
+		NSUrl WebUrl { get; }
+
+		[NullAllowed]
+		[Export ("artworkURL", ArgumentSemantic.Strong)]
+		NSUrl ArtworkUrl { get; }
+
+		[NullAllowed]
+		[Export ("videoURL", ArgumentSemantic.Strong)]
+		NSUrl VideoUrl { get; }
+
+		[Export ("explicitContent")]
+		bool ExplicitContent { get; }
+
+		[NullAllowed]
+		[Export ("isrc")]
+		string Isrc { get; }
+
+		[Static]
+		[Export ("mediaItemWithProperties:")]
+		SHMediaItem MediaItem (NSDictionary<NSString, NSObject> properties);
```

Comment:
Missing a verb, all methods in c# should start with one.

Reply:
Whoops! Good call!